### PR TITLE
Add EthereumForkArbiter

### PR DESCRIPTION
**Pull request by Recmo**

### CODE REVIEW

File: test/EthereumForkArbiter.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    await ethereumForkArbiter.signFork(block.number, block.hash, {
+      from: arbiter
+    });
+  });
+  it("should remember last signed fork", async () => {
+    const block = await web3.eth.getBlock("latest");
+    const tx = await ethereumForkArbiter.signFork(block.number, block.hash, {
+      from: arbiter
+    });
+    const rNumber = await ethereumForkArbiter.lastSignedBlockNumber.call();
```
Comment (krzkaczor) [inline test/EthereumForkArbiter.js]: is `r` coming from Remco? xD I think that standard testing vocabulary is to use "expected" for expected values and actual for ones coming from tested code.

File: test/EthereumForkArbiter.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    const rHash = await ethereumForkArbiter.lastSignedBlockHash.call();
+    const rTime = await ethereumForkArbiter.lastSignedTimestamp.call();
+    expect(rNumber).to.be.bignumber.equal(block.number);
+    expect(rHash).to.equal(block.hash);
+    const rBlock = await web3.eth.getBlock(tx.receipt.blockNumber);
```
Comment (krzkaczor) [inline test/EthereumForkArbiter.js]: how about following 3A principle? So do not interleave asserts with anything else?

File: test/EthereumForkArbiter.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        from: other
+      })
+    );
+    await ethereumForkArbiter.announceFork(name, url, {
+      from: arbiter
+    });
+  });
+  it("should only allow ROLE_FORK_ARBITER to sign", async () => {
+    const block = await web3.eth.getBlock("latest");
+    await expectThrow(
```
Comment (krzkaczor) [inline test/EthereumForkArbiter.js]: honestly, i would prefer to stick to expect + promised chai so something like:
`await expect(something).to.be.rejectedWith(EvmError); `

Comment (Recmo) [reply test/EthereumForkArbiter.js]: I like that too, but I want some abstraction over the specifics of the EVM error. It tends to change with clients and versions, check the `expectThrow` implementation to see what I mean.

File: test/EthereumForkArbiter.js (hunk 2 above)
Comment (Recmo) [reply test/EthereumForkArbiter.js]: Co to jest 3A principle?

File: test/EthereumForkArbiter.js (hunk 1 above)
Comment (Recmo) [reply test/EthereumForkArbiter.js]: “Received”. Will change to expected/actual

File: test/EthereumForkArbiter.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+contract("EthereumForkArbiter", ([deployer, arbiter, other]) => {
+  let ethereumForkArbiter;
+
+  beforeEach(async () => {
+    const accessPolicy = await createAccessPolicy([
+      { subject: arbiter, role: "ROLE_FORK_ARBITER" }
+    ]);
+    ethereumForkArbiter = await EthereumForkArbiter.new(accessPolicy);
+  });
+  it("should deploy", async () => {
```
[the remark this answers is not in the document]
Comment (Recmo) [reply test/EthereumForkArbiter.js]: Deblob the code. Roger.

File: contracts/EthereumForkArbiter.sol [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        public
+        only(ROLE_FORK_ARBITER)
+    {
+        require(block.blockhash(number) == hash);
+        lastSignedBlockNumber = number;
+        lastSignedBlockHash = hash;
+        lastSignedTimestamp = block.timestamp;
+        ForkSigned(lastSignedBlockNumber, lastSignedBlockHash);
+    }
+}
```
Comment (rudolfix) [inline contracts/EthereumForkArbiter.sol]: Conceptual problem: how this is intended to be used? Should we aggregate arbiter in each contract we care about forks? Like Neumark and LockedAccount?

Comment (Recmo) [reply contracts/EthereumForkArbiter.sol]: Just one globally should be enough, that's how I deploy it.

Comment (Recmo) [reply contracts/EthereumForkArbiter.sol]: Multiple would be a bad idea, they could only conflict.

Comment (Recmo) [reply contracts/EthereumForkArbiter.sol]: They don't need to be linked to the contracts. I did not want to put in a `kill all bad forks` mechanism for security reasons.

Only thing we need to do is put the contract address in the agreements in the paragraph that says how to identify the one-true-fork.

Comment (rudolfix) [reply contracts/EthereumForkArbiter.sol]: Yes, there should be just one, what I meant is that you keep its address in any contract that has agreement attached. In that case you have the same mechanism for legal and smart contract - a link to fork arbiter... We may think on adding arbiter with mixin that adds url to legal agreement. Let's discuss tomorrow

Comment (Recmo) [reply contracts/EthereumForkArbiter.sol]: I like the idea of connecting contracts back to the ForkArbiter. It also seems to overlap with the Agreement contract. I want to combine the two, so let's get this PR in and finish it in Agreement one, ok?

**Pull request merged**